x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/RBACEngine.java

```diff
@@ -524,9 +524,13 @@ private static boolean checkChangePasswordAction(Authentication authentication)
         }
 
         assert realmType != null;
-        // ensure the user was authenticated by a realm that we can change a password for. The native realm is an internal realm and
-        // right now only one can exist in the realm configuration - if this changes we should update this check
-        return ReservedRealm.TYPE.equals(realmType) || NativeRealmSettings.TYPE.equals(realmType);
+        // Ensure that the user is not authenticated with an access token or an API key.
+        // Also ensure that the user was authenticated by a realm that we can change a password for. The native realm is an internal realm
+        // and right now only one can exist in the realm configuration - if this changes we should update this check
+        final Authentication.AuthenticationType authType = authentication.getAuthenticationType();
+        return (authType.equals(Authentication.AuthenticationType.TOKEN) == false
+        && authType.equals(Authentication.AuthenticationType.API_KEY) == false )
```

Comment:
Why do we blacklist rather than whitelist here?
Shouldn't we just check for the positive `authType.equals(Authentication.AuthenticationType.REALM)` ?

Reply:
You're absolutely right